src/components/Pagination/index.jsx

```diff
@@ -32,29 +50,40 @@ const Pagination = ({
               className={cn("pagination-link", {
                 "is-active": number === pageIndex,
               })}
-              onClick={() => updatePageIndex}
-              onKeyDown={() => {}}
+              onClick={() => updatePageIndex(number)}
+              onKeyDown={() => updatePageIndex(number)}
               role="button"
               tabIndex="0"
             >
               {number}
             </a>
           ))}
-        <a className="pagination-ellipses">&hellip;</a>
-        <a
-          className="pagination-link"
-          onClick={() => updatePageIndex}
-          onKeyDown={() => {}}
-          role="button"
-          tabIndex="0"
-        >
-          {numRules}
-        </a>
+        {pageIndex <= Math.ceil(numRules / pageInterval - 5) && (
+          <span>
+            <a className="pagination-ellipses">&hellip;</a>
+            <a
+              className="pagination-link"
+              onClick={() =>
+                updatePageIndex(Math.ceil(numRules / pageInterval))
+              }
+              onKeyDown={() =>
+                updatePageIndex(Math.ceil(numRules / pageInterval))
+              }
+              role="button"
+              tabIndex="0"
+            >
+              {Math.ceil(numRules / pageInterval)}
+            </a>
+          </span>
+        )}
       </div>
       <a
-        className="btn btn-secondary btn-small pagination-next"
+        className={
+          `btn btn-secondary btn-small pagination-next ${ 
+          pageIndex > 25 ? "is-disabled" : ""}`
```

Comment:
although `25` is only used once, it's still nice to store it in a variable with a descriptive name for others or yourself in the future to remember what this number is supposed to represent